### PR TITLE
Bug 1337782 - Remove URI caching

### PR DESCRIPTION
**Pull request by denschub**
r? @MDTsai, see [bug 1337782](https://bugzilla.mozilla.org/show_bug.cgi?id=1337782)

### CODE REVIEW

File: src/content/lib/ua_overrider.jsm [hunk 1]
```diff
@@ -59,34 +58,7 @@ class UAOverrider {
 
   getUAForURI(uri) {
     let bareUri = uri.specIgnoringRef;
```
Comment (MDTsai) [inline src/content/lib/ua_overrider.jsm]: bareUri is not used, please remove unused variable.
As bareUri is removed, I think we don't need getUAForURI just for calling lookupUAOverride.

**Comment by denschub** [comment]
You are very right. Fixed. Also, rebased on latest master.

**Comment by MDTsai** [comment]
@denschub, I fail to run the unit test. Please also fix the test case then I think we are ready to merge the patch.

**Comment by denschub** [comment]
Not sure if you saw the IRC message, but this is fixed. :)

**Comment by denschub** [comment]
Oh, nvm, you r+'d on bugzilla. Should've read my emails first. Thanks for the review!

**Pull request merged**